sql/core/src/main/scala/org/apache/spark/sql/DataFrameImpl.scala

```diff
@@ -88,12 +88,24 @@ private[sql] class DataFrameImpl protected[sql](
     }
   }
 
-  protected[sql] def numericColumns: Seq[Expression] = {
-    schema.fields.filter(_.dataType.isInstanceOf[NumericType]).map { n =>
-      queryExecution.analyzed.resolve(n.name, sqlContext.analyzer.resolver).get
+  protected[sql] def numericColumns(colNames: String*): Seq[Expression] = {
+    val allNumbericCols = schema.fields.filter(_.dataType.isInstanceOf[NumericType]).map(_.name)
+    val diff = colNames.diff(allNumbericCols)
+    if (diff.nonEmpty) {
```

Comment:
One reason to filter all wrong given columns out is we can reports them at once.  I will fix it for the concern.